### PR TITLE
fixes issue with paired test

### PR DESCRIPTION
**Pull request by jwdebelius**
cleans up code (slightly)
corrects for pandas data typing and improves testing based on arrays
rather than lists

addresses #1436; this looks like a broken test following pandas compatibility

...I did have a bunch of test break associated with `skbio.io.format.tests.test_blast7.TestBlast7Reader`

### CODE REVIEW

File: skbio/stats/power.py [hunk 1]
```diff
@@ -1017,10 +1017,10 @@ def _identify_sample_groups(meta, cat, control_cats, order, strict_match):
         m_ids = meta.loc[ids].groupby(cat).groups
         # Checks if samples from the cat groups are represented in those
         # Samples
-        ids_vec = id_vecs = [m_ids[o] for o in order if o in
-                             m_ids]
+        id_vecs = [m_ids[o].values for o in order if o in
```
Comment (ebolyen) [inline skbio/stats/power.py]: Could you expand on why this change was made?  Was it a result of the pandas update? If so we may need to bump our minimum version of pandas.

File: skbio/stats/tests/test_power.py [hunk 2]
```diff
@@ -477,18 +477,23 @@ def test__identify_sample_groups(self):
 
     def test__identify_sample_groups_not_strict(self):
         # Defines the know values
-        known_pairs = {0: [['PP'], ['CD', 'NR']],
-                       1: [['MM', 'WM'], ['MH']],
-                       2: [['GW'], ['CB']]}
+        known_pairs = {1: [np.array(['PP'], dtype=object),
+                           np.array(['CD', 'NR'], dtype=object)],
+                       0: [np.array(['MM', 'WM'], dtype=object),
```
Comment (ebolyen) [inline skbio/stats/tests/test_power.py]: It looks like `0` and `1` are transposed from their previous mapping.

Comment (ebolyen) [inline skbio/stats/tests/test_power.py]: Should this be a list? it looks as if you are depending on iteration order of a dictionary.

File: skbio/stats/power.py (hunk 1 above)
Comment (ebolyen) [inline skbio/stats/power.py]: Sorry I mean adding `.values` to the call, should have clarified.

File: skbio/stats/tests/test_power.py (hunk 2 above)
Comment (jwdebelius) [inline skbio/stats/tests/test_power.py]: The 0 and 1 are proxies for a group description. The change in order reflects a change in the order returned by groupby (again, I think this is a pandas update result). The groups have remained consistent.

**Comment by RNAer** [comment]
@ebolyen , is there anything else blocking this PR?

File: skbio/stats/power.py (hunk 1 above)
Comment (ebolyen) [inline skbio/stats/power.py]: Just tested this PR with a couple of different pandas versions. If we want to avoid bumping our minimum pandas version, we need to drop `.values`. Older versions of pandas return lists here apparently, and index objects are essentially ndarrays as it is, so it works fine without.

File: skbio/stats/tests/test_power.py (hunk 2 above)
Comment (ebolyen) [inline skbio/stats/tests/test_power.py]: Reverting the order with older versions causes the tests to fail. I honestly have no idea how this was _ever_ passing... Good to leave as is, thanks.

**Comment by jwdebelius** [comment]
@ebolyen Pinging the updated review request

**Comment by ebolyen** [comment]
Merged, thanks @jwdebelius!
